targets/TARGET_STM/TARGET_STM32G4/TARGET_STM32G474xx/TARGET_NUCLEO_G474RE/system_clock.c

```diff
@@ -105,9 +105,9 @@ uint8_t SetSysClock_PLL_HSE(uint8_t bypass)
     RCC_OscInitStruct.PLL.PLLState = RCC_PLL_ON;
     RCC_OscInitStruct.PLL.PLLSource = RCC_PLLSOURCE_HSE;
     RCC_OscInitStruct.PLL.PLLM = RCC_PLLM_DIV6;
-    RCC_OscInitStruct.PLL.PLLN = 85;
+    RCC_OscInitStruct.PLL.PLLN = 80;
     RCC_OscInitStruct.PLL.PLLP = RCC_PLLP_DIV2;
-    RCC_OscInitStruct.PLL.PLLQ = RCC_PLLQ_DIV2;
+    RCC_OscInitStruct.PLL.PLLQ = RCC_PLLQ_DIV4;
```

Comment:
So you decrease SystemCoreClock from 170 MHz to 160,
and you get PLLQ frequency at 80 MHz

Could you explain me again the reason ?
Thx

Reply:
With a core frequency of 170MHz, many CAN-frequencies in my tests could not be setup correctly. 

For example I worked at a frequency of 250k. With a core frequency of 170MHz this gave me:

| Field | Calculated Value | Register Value |
|------|-------|-----|
| ntq1 	| 510 (0x1FE)	| 253 (0xfd) |
| ntq2 	| 169 (0xA9)	| 40  (0x28) |
| sjw 	| 169 (0xA9)	| 40  (0x28) |

So the register could not store the value, the interface could not be setup properly.

The current limitation is the result of the api not using the prescaler (always at 1). 

I propose to use the prescaler, calculated this way:
```C
    int ntq = HAL_RCCEx_GetPeriphCLKFreq(RCC_PERIPHCLK_FDCAN) / hz;

    int nominalPrescaler = 1;
    // !When the sample point should be lower than 50%, this must be changed to
    // !IS_FDCAN_NOMINAL_TSEG2(ntq/nominalPrescaler), since
    // NTSEG2 and SJW max values are lower. For now the sample point is fix @75%
    while (!IS_FDCAN_NOMINAL_TSEG1(ntq/nominalPrescaler)){
        nominalPrescaler ++;
        if (!IS_FDCAN_NOMINAL_PRESCALER(nominalPrescaler)){
            error("Could not determine good nominalPrescaler. Bad clock value\n");
        }
    }
    ntq = ntq/nominalPrescaler;
```
See commit. 

Still there can't be guaranteed, that the resulting timing is within tolerance, so this will still require manual adaptation when using a strange value.

Even with that a frequency of 170MHz caused many setups to fail, so I reduced it to 80MHz, which worked just fine for most (not all) of my test-configurations. Unfortunately I not record my tests properly, I can repeat and record them to show it better, if needed.

Reply:
Thx for details.
Maybe we could code like this:
````
#if DEVICE_CAN
PLLN = 80
#else
PLLN = 85
````
If you disable CAN, you keep the maximum capability ?

Reply:
That fine by me, I will add this with a comment.

Reply:
Added in 2a13fa199dcc9f2a8faef79b7377e12222830d23, added comment in 73493b909a442d97f1d208c14620ca8d55636223.